src/parachain/issue.ts

```diff
@@ -121,33 +159,73 @@ export class DefaultIssueAPI extends DefaultTransactionAPI implements IssueAPI
         this.feeAPI = new DefaultFeeAPI(api);
     }
 
+    async getRequestLimits(vaults?: Map<AccountId, Big>): Promise<IssueLimits> {
+        if (!vaults) vaults = await this.vaultsAPI.getVaultsWithIssuableTokens();
+        const [singleVaultMaxIssuable, totalMaxIssuable] = [...vaults.entries()].reduce(
+            ([singleVault, maxTotal], [_, vaultAvailable]) => {
+                maxTotal = maxTotal.plus(vaultAvailable);
+                singleVault = singleVault.gt(vaultAvailable) ? singleVault : vaultAvailable;
```

Comment:
The `getVaultsWithIssuableTokens()` is sorted so to get the maximum for a single vault, you can just take the top element.